CoffeeChat/Controller/EditingViewController.swift

```diff
@@ -233,15 +254,76 @@ class EditingViewController: UIViewController {
         navigationItem.rightBarButtonItem = saveBarButtonItem
     }
 
+    // MARK: - Button Action
     @objc private func backPressed() {
         navigationController?.popViewController(animated: true)
     }
 
     @objc func savePressed() {
-        // TODO save favorites
+        if let yourSection = yourSection {
+            var interests: [Interest] = []
+            var groups: [Group] = []
+
+            for item in yourSection.items {
+                switch item {
+                case .interest(let interest):
+                    interests.append(interest)
+                case .group(let group):
+                    groups.append(group)
+                }
+            }
+
+            if interests.count > 0 {
+                NetworkManager.shared.updateUserInterests(interests: interests.map({ $0.name })).observe { result in
+                    switch result {
+                    case .value(let response):
+                        if response.success {
+                            print("Interests updated successfully")
+                        } else {
+                            print("Interests couldn't be updated")
+                        }
+                    case .error:
+                        print("Networking error when trying to update interests")
+                    }
+                }
+            }
+
+            if groups.count > 0 {
+                NetworkManager.shared.updateUserGroups(groups: groups.map { $0.name }).observe { result in
+                    switch result {
+                    case .value(let response):
+                        if response.success {
+                            print("Groups updated successfully")
+                        } else {
+                            print("Groups couldn't be updated")
+                        }
+                    case .error:
+                        print("Networking error when trying to update groups")
+                    }
+                }
+            }
+        }
         navigationController?.popViewController(animated: true)
     }
 
+    /// Remove duplicated entries in `yourStrings` and `moreStrings`
+    private func removeDuplicates(yourStrings: [String], moreStrings: [String]) -> (your: [String], more: [String]) {
+        var set = Set<String>()
+
+        yourStrings.forEach { set.insert($0) }
+        let yourList = Array(set).sorted()
+
+        var moreList: [String] = []
+        for string in moreStrings {
+            if set.insert(string).inserted {
+                moreList.append(string)
+            }
+        }
```

Comment:
It seems to me that this is both adding to `moreList` only those strings in `moreStrings` that are not in `yourStrings` and removing duplicates. If this is correct, I think this "mutual exclusion" should be documented.